### DISCUSSION TITLE
AdonisJS community marketing ideas and efforts

### DISCUSSION THREAD

**Discussion by bitkidd** [adonisjs/core#3281] [Ideas]
AdonisJS is a great framework created by @thetutlage and core members. An awesome tool that makes developers and businesses happier. It has a nice community that may help it grow faster, become more popular and popularise the ideas it brings even more, but this needs some work to be done by everyone, of course on a volunteer bases.

The ideas that may help long term with some marketing goals:
1. **Merch and stickers**. Everyone love stickers, especially the ones that are limited in edition.
2. **Email newsletter should be a regular one**. It may require a publishing plan, but a regular newsletter with news, future plans, tips and tricks is important, regularity here is the key as it may allow the community to understand where everything goes.
3. **Tighter work with a community**. Of course core members may not have the time to write articles and posts here and there to promote the framework and share knowledge, but they may help the community to define some mandatory topics, I guess there are people, including me, who can write those posts. At the same time we can create a hub that will include or at least mention those posts on a AdonisJS blog domain for SEO purposes, as DigitalOcean did with their community website with multiple tutorials and etc.
4. **AdonisJS Community organisation**. There is an AdonisJS Community organisation here on GitHub, maybe we can invite people to transfer repos to it, so everyone would have a badge, people love badges and they may help to promote the framework. A nice example of that kind of stuff is [React Native Community](https://github.com/react-native-community) organisation.
5. **Patreon supporters**. It could be a real motivation for people to support the framework if they could get something from it, a private discord server or channels, a private package for those on a higher tier and etc.
6. **YouTube, Twitch**. It may have a potential positive impact to create AdonisJS YouTube and Twitch channels that will combine multiple other sources that are ready to contribute videos for these channels. At least it will be easier to google official sources.

These are my ideas about things that may help to promote the framework and make the community shine even more.

This discussion is started [here](https://twitter.com/romainlanz/status/1452629004004044810).

**Comment by thetutlage** [comment]
Alright. I was waiting to see if someone else wanted to add something to the conversation. So let me start by responding to the @bitkidd initial comment.

- **Merch and stickers**: Doable, and we should do. It's just a low priority for me right now. Because I am not sure if people pay for this stuff + shipping cost. Usually, stickers and t-shirts do well when you give them for free at some conference. 

- **Email newsletter should be a regular one**- We already have one https://news.adonisjs.com, and I am consistent with it so far ( the sample size is small, though ). Doing it weekly doesn't make much sense because we do not have meaningful updates that frequently. And I am not in favor of just sending out anything in the name of a weekly newsletter.

- **Tighter work with a community** - As you mentioned, I have significantly less time to write articles right now. Happy to work with the community on the same. **If 5-6 people can volunteer to write content, I can do a monthly meet-up with you guys, and we can discuss next month's focus + roadmap in that cal**l. Also, I am a little hesitant here because I have had experiences where people disappear after volunteering.

- **AdonisJS Community organisation** - Okay. How does it work? Are people still the owners of their repos and they publish to npm from their account? Because owning someone else's work is the last thing I want to do.

- **Patreon supporters** - Here, I prefer to create a content website and charge for it directly as a monthly membership.

- **YouTube, Twitch** - Agree.

----

Apart from the above points, I am really curious to know "What is stopping you from using AdonisJS personally or at work".

I am asking this because I think "chasing being popular" is a wrong goal to start with. But addressing concerns that are stopping people from using AdonisJS at work is worth it.

There are some things I can address by working closely with the community, like addressing some technical flaws, creating learning content, presenting AdonisJS in some meet-ups and conferences. 

However, some things can be never be fixed because they are not problems in the first place. For example, "AdonisJS is a not a project from a big tech company like Google or Facebook", "Or, AdonisJS doesn't have evangelist cheering for it every morning". 

I know the Open Source space has changed a lot recently. VCs are pouring cash, and big tech companies are using it as a sales funnel. But, being an independent open source creator is not a bad or inferior thing. That's how Open source got started in the first place, and I am way happier working independently on it vs. taking cash from someone and creating artificial growth + hype.


---

To finish. AdonisJS is more than a side project for me. I am creating a living out of it right now and will continue to explore ways to monetize it. So I want AdonisJS to do well and add value to people's lives in real. So feel free to share your thoughts if you think you badly want to use AdonisJS, but your employer has some issues that I can address.  "No MongoDB support is not an issue 😛"

**Reply by bitkidd** [reply]
**About AdonisJS Community and motivation.** 

The idea to invite package creators to the parent organisation came to my mind as a reply to a very bad situation in js world with ecosystem and its segmentation, there are myriads of packages that are very important and just left over with no maintainers, no support and no updates, forcing forks on forks situation, duplicating packages. In my mind combining packages under one org may allow to mitigate these and some other problems. 

The workflow may look like this: 
- Package author is invited (or may ask just right away in discord or anywhere) to move his repo to the parent org and becomes its maintainer with all the rights to do with it whatever he wants, as the code belongs to the original author
- At the same time this same package may be published under relative npm org, npm organisations allow role management per org and per package
- If a package owner does not want to support it anymore, organisation managers may ask community to take this package and assign someone else who wants to work on it, basically organisation structure may allow to keep the ecosystem more stable and much more trustable

This may positively impact the "feel" of the framework, as not only the core packages will be managed, but the addons too.

**Comment by Barbapapazes** [comment]
Hello,


I'm a french student working with Adonis for a big student life side project. I really like this framework and I believe in it very sincerely.

I wrote a tutorial a few months ago and I will be happy to propose my ideas and exchange with you to work on promoting the framework.

I think there is a real opportunity to create real-world content, like the polls-app can be, and to explain it! I also think that we should maybe go to the content creators on YouTube to make tutorials.

**Comment by thetutlage** [comment]
Writing it here separately (in case someone wasn't interested in reading huge wall of text).

I am happy to setup bi-weekly meetings with individuals who want to become part of a closer group and contribute to the framework by regularly writing content, doing screencasts, giving talks or presenting AdonisJS in a meet up. Basically, anyone who is a die hard AdonisJS fan and want to spread the word by teaching and helping others

**Reply by ndianabasi** [reply]
I'm interested in this. I think I can champion text-based content creation.

**Comment by McSneaky** [comment]
That organization idea seems cool :) 

I'll start to push some friend designers again to move on with the merch. If not getting any response / progress within some days will announce "looking for designer" in Discord or will try to look from Fiverr

(Aah, just realized I'm going on vacation at the end of this week for two weeks.. So I'll give designers about two weeks 🙂)

> Tighter work with a community - ...  Also, I am a little hesitant here because I have had experiences where people disappear after volunteering.

I think having small group in here would be better. One by one people might fall off, but when there's small community group / team, then there's more team feeling and more sticking together. Maybe some Discord channel for this group? Also when there's small group then it doesn't matter that much if one person falls off, rest can handle 🙂 Ties well with that organization idea

> Email newsletter should be a regular one

I think that's good idea too, but it can be something like once a month and maybe just some links from there to "Community creations" and two sentences about progress and plans for close future (next some weeks / months). Every weekly will get too spammy with just "Updated packages", "No progress", "Mezie wrote new blog post" etc

> I am happy to setup bi-weekly meetings with individuals who want to become part of a closer group and contribute to the framework by regularly writing content, doing screencasts, giving talks or presenting AdonisJS in a meet up. Basically, anyone who is a die hard AdonisJS fan and want to spread the word by teaching and helping others

"Die hard AdonisJS fan" is my middle name! 😄

**Reply by McSneaky** [reply]
Designs status update:

> I'll start to push some friend designers again to move on with the merch. If not getting any response / progress within some days will announce "looking for designer" in Discord or will try to look from Fiverr

One told she'll pick it up next week, other haven't responded yet. Most likely too loaded with work already or smth. Waiting for end of next week and then will ping her again to see. If still nothing I'll ask from Discord if we have anyone who can do designs or check into Fiverr (or similar)

**Comment by ammezie** [comment]
@McSneaky guess we both have the same middle name 🤣.

I think the once a month newsletter is cool for now considering there really isn't much going on in the community for now. Though incorporating @McSneaky suggestions will make is more cool and give better insight.

Most times, I wish I could do more and get involved more the community. @ndianabasi and I are working on a community outreach thingy to spread the word about AdonisJS.

**Comment by ndianabasi** [comment]
I wasn't aware of this thread as I don't frequent GitHub discussions.

I love the ongoing discussions and appreciate @bitkidd for initiating it. 

As @ammezie affirmed, we are meeting this evening to kick-start discussions about building the AdonisJS Community in Africa.

Let's forget about the experienced Nodejs developers who won't let go of Express and other frameworks. Let's target the new developers who are just learning Nodejs for the first time. If we capture that segment of developers, then AdonisJS is set to grow exponentially in the next 3 to 5 years when these developers become junior/senior engineers.

At the same time, we will do more to soft-sell AdonisJS to companies through conferences and partnerships with content creators and frontend frameworks. I've talked about partnership with frontend frameworks because without them AdonisJS won't reach its full potential.

We have to deliberately reach out to them. Let the Core Team market AdonisJS Server module as a replacement for ExpressJS. The frontend frameworks can then begin to experiment with swapping Express with AdonisJS Server + middlewares when developing their SSR capabilities.

I've learnt from watching Vite take over the Build Tools niche. They didn't achieve that by focusing on Vuejs alone. They made Vite framework agnostic. Aside from AdonisJS being used mainly for API services, I think if we begin to make it frontend friendly by exposing the AdonisJS Server for integrations into SSR, it will draw the attention of the frontend development players who can then begin to integrate AdonisJS modules into their projects.

For me, it is possible to make Strapi, Nuxtjs, etc. use AdonisJS modules/hooks. AdonisJS is very modular and has the right hooks that those framework developers need.

**Reply by eokwukwe** [reply]
I totally agree with the above suggestions.

**Comment by McSneaky** [comment]
Status update about designs

Found one more designer from Discord who told will do some research, gather ideas etc and will get back in some days :) 

If you know someone who does designs then would be cool to bring us together :) We can always have designs by multiple designers (or in case someone falls off)

**Comment by eokwukwe** [comment]
Having worked with Ruby on Ralls (though briefly) and currently working with Laraval, I was excited when I discovered AdonisJS and have been following it since the V5 preview. I have always wanted to see a framework like this for Node.js (_**I know the pain of working with Express.js**_). And I have tried, in my little way, to preach AdonisJS to others whenever I have the opportunity.

For me, the only reason I am not yet working with AdonisJS is that I do not currently have any work project that uses AdonisJS. I am past the era of building dummy projects (no time for that). _However, I am currently trying out a concept of a meetup app for AdonisJS developers. I call it [Adonievents](https://github.com/eokwukwe/adonievents) (using AdonisJS and Inertia)_.  I will be happy to join any company/project that uses AdonisJS (paid or open-source). And @ammezie and @ndianabasi I am interested in any initiative to help promote AdonisJS in Africa.

**Reply by ndianabasi** [reply]
Hi @eokwukwe. Thank you for your interest in AdonisJS Africa. Please can you inbox me on Twitter so that I can send you the invite? Thank you.

**Reply by eokwukwe** [reply]
@ndianabasi Twitter is officially banned in Nigeria for now, so I can't assess it. However, here is my email okwukwe.ewurum@gmail.com
